### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by gammazero**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,30 @@
 # go-indexer-core
+[![Go Reference](https://pkg.go.dev/badge/github.com/filecoin-project/go-indexer-core.svg)](https://pkg.go.dev/github.com/filecoin-project/go-indexer-core)
```
Comment (willscott) [inline README.md:2]: are any of the other badges from https://github.com/ipld/go-car/blob/master/README.md worth copying?

File: README.md [hunk 2]
```diff
@@ -1 +1,30 @@
...
+> Storage specialized for indexing provider content
+
+This is a storage system for storing and retrieving the mapping of content multihashes to provider data.  It is optimized for storing large numbers of multihashes mapping to relatively few provider data objects.  Provider data describes the providers of content referenced by a multihash and information on how to retrieve that content from the providers.  The data to store is given as a provider value object and a set of multihashes that are provided by that provider using that value object.  Data is retrieved by using a multihash to lookup the provider value objects.  Provider data can be updated and removed independent of the multihashes that map to it.
+
+### Configurable Cache
+An integrated cache is included to aid in fast index lookups.  The cache can be optionally disabled, and its size is configurable. The cache interface allows other cache implementations to be used.
```
Comment (willscott) [inline README.md:8]: how do you configure it?

File: README.md [hunk 3]
```diff
@@ -1 +1,30 @@
...
+
+### Choice of Persistent Storage
+The persistent storage is provided by a choice of storage systems that include [storethehash](https://github.com/ipld/go-storethehash), [pogrep](https://github.com/akrylysov/pogreb#readme), and an in-memory implementation.  The storage interface allows any other storage system solution to be adapted.
```
Comment (willscott) [inline README.md:11]: how do you choose it?

**Review by willscott — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (gammazero) [reply README.md:3]: Probably, added some.

File: README.md (hunk 2 above)
Comment (gammazero) [reply README.md:10]: Shown in example

File: README.md (hunk 3 above)
Comment (gammazero) [reply README.md:15]: Shown in example

**Pull request merged**